hook/lua/shield.lua

```diff
@@ -0,0 +1,944 @@
+
+                    -- keep track of overspill damage if we have not received any actual damage yet
+                    self.DamagedOverspill[instigatorId] = self.DamagedOverspill[instigatorId] + amount
+                end
+            end
+        end
+
+        -- do damage logic for shield
+
+        if self.Owner ~= instigator then
+            local absorbed = self:OnGetDamageAbsorption(instigator, amount, dmgType)
+
+            LOG("We're calculating the damage now... - ApplyDamage Function")
+            LOG("Absorbed is "..repr(absorbed))
+
+            -- take some damage
+            EntityAdjustHealth(self, instigator, -absorbed)
+
+            -- check to spawn impact effect
+            local r = Random(1, self.Size)
+            if dmgType ~= "ShieldSpill"
+                and not (self.LiveImpactEntities > 10
+                    and (r >= 0.2 * self.Size and r < self.LiveImpactEntities))
+            then
+                ForkThread(self.CreateImpactEffect, self, vector)
+            end
+
+            if self.RegenThread then
+                KillThread(self.RegenThread)
+                self.RegenThread = nil
+            end
+             
+            if GetHealth(self) <= 0 then
+                ChangeState(self, self.DamageRechargeState)
+            else
+                if self.OffHealth < 0 then
+                 
+                    ForkTo(self.CreateImpactEffect, self, vector)
+                     
+                    if self.RegenRate > 0 then
+                     
+                        self.RegenThread = self:ForkThread(self.RegenStartThread)
+     
+                    end
+                else
+                    self:UpdateShieldRatio(0) 
+                end
+            end
+        end
+
+        -- overspill damage checks
+
+        LOG("doOverspill is "..repr(doOverspill))
+
+        if -- prevent recursively applying overspill
+        doOverspill
+            -- personal shields do not have overspill damage
+            and self.ShieldType ~= "Personal"
+            -- we consider damage without an instigator irrelevant, typically force events
+            and IsEntity(instigator)
+            -- we consider damage that is 1 or lower irrelevant, typically force events
+            and amount > 1
+            -- do not recursively apply overspill damage
+            and dmgType ~= "ShieldSpill"
+        then
+            local spillAmount = self.SpillOverDmgMod * amount
+
+            LOG("The spillAmount is "..repr(spillAmount))
+
+            -- retrieve shields that overlap with us
+            local others, count = self:GetOverlappingShields(tick)
+
+            LOG("What is the Overlapping Shield Amount (others) "..repr(others).. "and Count is "..repr(count))
+
+            -- apply overspill damage to neighbour shields
+            for k = 1, count do
+                others[k]:ApplyDamage(
+                    LOG("Applying Overspill Damage to other shields nearby"),
```

Comment:
This log might be the culprit

Reply:
Should be fixed now, still working on figuring out why the overspill isnt applying to shields.